### PR TITLE
[NTDLL_APITEST] Nt*InformationProcess: Fix 1 check and improve some others

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Let these 2 tests fully succeed on WS03.

## Proposed changes

- Fix
'probelib.c:...: Test failed: 0xc0000004 or special status (0xc0000061) expected but got 0xc0000005 for class information 16 in set information process operation!'
- Be explicit about allocation failures.
- Remove redundant 1-aligned checks.
- Better set/check parameters.

### CODE REVIEW

File: modules/rostests/apitests/ntdll/NtQueryInformationProcess.c [hunk 1]
```diff
@@ -209,30 +211,14 @@ Test_ProcessPriorityClassAlignment(void)
     RtlFillMemory(&ProcPriority->PriorityClass, sizeof(ProcPriority->PriorityClass), 0x55);
 
     /* Unaligned buffer -- wrong size */
-    Status = NtQueryInformationProcess(NtCurrentProcess(),
```
Comment (learn-more) [inline modules/rostests/apitests/ntdll/NtQueryInformationProcess.c:212]: why did you remove those tests?

Comment (SergeGautherie) [reply modules/rostests/apitests/ntdll/NtQueryInformationProcess.c:212]: 2-aligned checks fail, so 1-aligned checks seem to be useless.

File: modules/rostests/apitests/ntdll/NtQueryInformationProcess.c [hunk 2]
```diff
@@ -270,12 +257,14 @@ Test_ProcessWx86Information(void)
...
 
     /* Given an invalid process handle */
+    VdmPower = 0xDEADBEEF;
     Status = NtQueryInformationProcess(NULL,
                                        ProcessWx86Information,
                                        &VdmPower,
                                        sizeof(VdmPower),
                                        NULL);
     ok_hex(Status, STATUS_INVALID_HANDLE);
+    ok_dec(VdmPower, 0xDEADBEEF);
```
Comment (GeoB99) [inline modules/rostests/apitests/ntdll/NtQueryInformationProcess.c:267]: That doesn't make much sense to me. Whenever `NtQueryInformationProcess` fails for whatever reason, such as in this case, the submitted variable that had to get the queried information doesn't change simply because the function halted during the moment the failure has been constrained, unless you have discovered that Windows does something to that variable whilst the function bails out.

File: modules/rostests/apitests/ntdll/NtSetInformationProcess.c [hunk 3]
```diff
@@ -30,11 +30,13 @@ Test_ProcBasePriorityClass(void)
     ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
 
     /* Set the base priority to an invalid process handle */
+    BasePriority = HIGH_PRIORITY;
     Status = NtSetInformationProcess(NULL,
                                      ProcessBasePriority,
                                      &BasePriority,
                                      sizeof(KPRIORITY));
     ok_hex(Status, STATUS_INVALID_HANDLE);
+    ok_dec(BasePriority, HIGH_PRIORITY);
```
Comment (GeoB99) [inline modules/rostests/apitests/ntdll/NtSetInformationProcess.c:39]: Ditto.
